src/App.tsx

```diff
@@ -1,61 +1,133 @@
+import { useState, ChangeEvent } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { ToDo } from './types/types';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const findUserById = (userId:number | null) => {
+  return usersFromServer.find(user => user.id === userId) || null;
+};
+
+const extendedTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUserById(todo.userId),
+} || null));
 
 export const App = () => {
+  const [todos, setTodos] = useState<ToDo[]>(extendedTodos);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setselectedUserId] = useState<number | null>(null);
+
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (titleError !== '') {
```

Comment:
Here it would be better to check for an empty `title` because now I can create a new todo with an empty title.